lib/hash.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var bcrypt = require('bcrypt');
+
+var except = require('./utils').except;
+
+function addToQueue (config) {
+  return function (queue, field) {
+    var fieldConfig = config.fields[field];
+    var needsHash = fieldConfig.hash === true || fieldConfig.password === true;
+
+    return needsHash ? queue.concat(field) : queue;
+  };
+}
+
+function successiveHash (data, hashed, hashQueue, cb) {
+  var toBeHashed = hashQueue.pop();
+
+  if (typeof toBeHashed === 'undefined') {
+    return cb(null, hashed);
+  }
+
+  return bcrypt.hash(data[toBeHashed], 10, function (err, hash) {
```

Comment:
Ok. so we are _hard-coding_ the `bcrypt` hashing "_cycles_" in the library.
also, how are we going to _use_ this hashed data?
if we attempt to `bcrypt.compare` we need to have the plaintext 
see: https://github.com/ncb000gt/node.bcrypt.js/#to-check-a-password